HDDS-3947: Sort DNs for client when the key is a file for #getFileStatus #listStatus APIs

rakeshadr opened the pull request:
## What changes were proposed in this pull request?

Similar to OzoneManagerFS#lookupFile(OmKeyArgs args, String clientAddress) interface, "clientAddress" has been passed as an argument to OzoneManagerFS#listStatus(), OzoneManagerFS#getFileStatus() APIs.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3947

## How was this patch tested?

Used existing TestKeyManagerImpl unit test cases.

bshashikant commented on hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java:
```diff
@@ -1016,6 +1016,7 @@ public OzoneFileStatus getOzoneFileStatus(String volumeName,
         .setBucketName(bucketName)
         .setKeyName(keyName)
         .setRefreshPipeline(true)
+        .setSortDatanodesInPipeline(topologyAwareReadEnabled)
```
@rakeshadr , can you plz add some details here why sorting of dns is required here?

rakeshadr replied:
Thanks @bshashikant for the reviews. This change is basically to make "getFileStatus" and "listStatus" apis similar to "lookupFile()" API, where it does sorting.

Reference:
https://github.com/apache/hadoop-ozone/pull/1164#discussion_r451122204

adoroszlai reviewed (commented):
Thanks @rakeshadr for working on this.  Can you please update the patch to resolve conflicts?

rakeshadr commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/fs/OzoneManagerFS.java:
```diff
@@ -31,7 +31,18 @@
+   * @param clientAddress a hint to key manager, order the datanode in returned
+   *                      pipeline by distance between client and datanode.
+   * @return file status.
+   * @throws IOException
```
No specific reason, I will describe exception cases. 
I will update the KeyManagerImpl implementation as well.

rakeshadr commented on hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/fs/OzoneManagerFS.java:
```diff
@@ -31,7 +31,18 @@
+   * @return file status.
+   * @throws IOException
+   */
+  OzoneFileStatus getFileStatus(OmKeyArgs args, String clientAddress)
```
Agreed, will keep the existing #getFileStatus(args) method.

rakeshadr reviewed (commented):
Thanks @adoroszlai for the reviews

elek commented:
I am afraid something wrong happened with the last rebase/merge:  

```
Files changed 737
```

rakeshadr commented:
@bshashikant , @adoroszlai I have created fresh PR https://github.com/apache/hadoop-ozone/pull/1385 as this PR has been polluted:-)

I have addressed review comments in that new PR.
